piano-hero/mobile/src/main/java/com/novoda/pianohero/GameMvp.java

```diff
@@ -0,0 +1,38 @@
+package com.novoda.pianohero;
+
+interface GameMvp {
+
+    interface Model {
+
+        RoundViewModel startGame();
+
+        RoundViewModel onNotesPlayed(Callback callback, Note... note);
+
+        interface Callback {
+            void onSequenceComplete();
+        }
+    }
+
+    interface View {
+
+        void showRound(RoundViewModel viewModel);
+
+        void showGameComplete();
+
+        void showError();
+
+        void showKeyboard();
```

Comment:
what if I told you this was a _keyboard_

![](http://www.agnes-bruckner.com/apronus_images/facepiano.gif)